weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx

```diff
@@ -595,6 +656,119 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
+  //   dim: PlotState.DimensionLike;
+  // };
+
+  // const extractWeaveExpressionDims = (dim: PlotState.DimensionLike) => {
+  //   console.log('dimension.name', dim.name);
+  //   if (PlotState.isWeaveExpression(dim)) {
+  //     return [dim];
+  //   } else if (PlotState.isDropdownWithExpression(dim)) {
+  //     const newDim =
+  //       dim.mode() === 'expression' ? dim.expressionDim : dim.dropdownDim;
+  //     return [newDim];
+  //   } else if (PlotState.isGroup(dim)) {
+  //     return dim.activeDimensions();
+  //   }
+  //   return [];
+  // };
+
+  // // groupByDom
+  // // I need to grab the list of dimensions.
+  // const groupByDims = useMemo(() => {
+  //   // for each base dimension, if it's expanded
+  //   // then we need to add to the list
+  //   const dimensions: GroupDim[] = [];
+  //   for (const dimName of PLOT_DIMS_UI) {
+  //     console.log('dimName', dimName);
+  //     const dimIsShared = PlotState.isDimShared(config.series, dimName, weave);
+  //     const dimIsExpanded = config.configOptionsExpanded[dimName];
+  //     const dimIsSharedInUI = dimIsShared && !dimIsExpanded;
+
+  //     const dimObject = PlotState.dimConstructors[dimName](
+  //       config.series[0],
+  //       weave
+  //     );
+  //     console.log('dimIsSharedInUI', dimIsSharedInUI);
+  //     if (dimIsSharedInUI) {
+  //       extractWeaveExpressionDims(dimObject).map(dim => {
+  //         dimensions.push({name: dimName, dim});
+  //       });
+  //     } else {
+  //       for (let i = 0; i < config.series.length; i++) {
+  //         const seriesDim = PlotState.dimConstructors[dimName](
+  //           config.series[i],
+  //           weave
+  //         );
+  //         extractWeaveExpressionDims(seriesDim).map(dim => {
+  //           dimensions.push({name: `${dimName}-${i + 1}`, dim});
+  //         });
+  //       }
+  //     }
+  //   }
+  //   return dimensions;
+  // }, [config, weave]);
+
+  // const configGroupedDims = useMemo(() => {
+  //   const groupedDims: string[] = [];
+  //   config.series.map((s, i) => {
+  //     s.table.groupBy.map(col => {
+  //       for (let key in s.dims) {
+  //         const value = s.dims[key as keyof DIM_NAME_MAP] as string;
+  //         if (value === col) {
+  //           groupedDims.push(`${key}-${i + 1}`);
+  //         }
+  //       }
+  //     });
+  //   });
+  //   return groupedDims;
+  // }, [config]);
+
+  // const groupByDom = useMemo(() => {
+  //   return (
+  //     <ConfigPanel.ConfigOption label="Grouping">
+  //       <ConfigPanel.ModifiedDropdownConfigField
+  //         multiple
+  //         options={groupByDims.map(d => {
+  //           return {key: d.name, text: d.name, value: d.name};
+  //         })}
+  //         value={configGroupedDims}
+  //         onChange={(event, {value}) => {
+  //           console.log('hey dropdown value set by user', value);
+  //         }}
+  //       />
+  //     </ConfigPanel.ConfigOption>
+  //   );
+  // }, [groupByDims, config]);
```

Comment:
we'll want this in the diff for the next pr

Reply:
I'm going to work on this right after this PR so I thought it'd be easier just to have the commented out portion there for easy access

Reply:
can you just make a new branch with this in there? otherwise when I look at your next PR there wont be a diff. and i wont know what went in related to grouping without manually looking in the code

Reply:
nvm. i guess there will be a diff with the comment // removed

Reply:
moved into new branch